src/version/plugins/Feeds/VersionSummarySelectors.js

```diff
@@ -0,0 +1,110 @@
+import React, { Component } from "react";
+import { connect } from 'react-redux';
+
+import Typography from "@material-ui/core/Typography";
+import Divider from '@material-ui/core/Divider';
+import Checkbox from '@material-ui/core/Checkbox';
+
+import { feedsListAction } from "../../actions/feedsListAction";
+import { setSelectedFeedsAction } from "../../actions/setSelectedFeedsAction";
+import { feedsRssAction } from "../../actions/feedsRssAction";
+
+import dummyFeeds from "./dummyFeeds";
+
+/**
+ * This component returns selectors for Feeds
+ *
+ * @author Bogdan Shcherban <bsc@piogroup.net>
+ */
+class VersionSummarySelectors extends Component {
+
+    state = {
+        selectedFeeds: this.props.selectedFeeds ? Object.values(this.props.selectedFeeds) : [],
+    };
+
+    componentDidMount() {
+        this.props.getUserFeeds();
+    };
+
+    /**
+     * This function checks is current Feed was selected
+     *
+     * @author Bogdan Shcherban <bsc@piogroup.net>
+     * @param {string} sourceId
+     * @return {boolean}
+     */
+    isFeedsChecked = sourceId => {
+        const { selectedFeeds } = this.state;
+        return selectedFeeds.indexOf(sourceId) !== -1;
+    };
+
+    /**
+     * This function toggle Feeds visibility
+     *
+     * @author Bogdan Shcherban <bsc@piogroup.net>
+     * @param {shape} item
+     */
+    toggleVisibility = item => {
+        this.setState(state => {
+                const { selectedFeeds } = state;
+                let feedsArray = selectedFeeds;
+                if (Object.values(selectedFeeds).indexOf(item.sourceId) !== -1) {
+                    let index = selectedFeeds.indexOf(item.sourceId);
+                    feedsArray.splice(index, 1);
+                } else {
+                    feedsArray.push(item.sourceId);
+                    this.props.getRssData(item.sourceId, item.rssFeedUrl);
```

Comment:
you should do it in the callback after the setState, not during the setState

Reply:
Done. This function is called in callback instead of setState